test/common/network/udp_listener_impl_test.cc

```diff
@@ -397,6 +421,139 @@ TEST_P(UdpListenerImplTest, SendDataError) {
   EXPECT_DEATH(listener_->send(send_data), "Invalid argument passed in");
 }
 
+/**
+ * Test that multiple stacked packets of the same size are properly segmented
+ * when UDP GRO is enabled on the platform.
+ *
+ * [Only run the following test if UDP_GRO is configured]
+ **/
+#ifdef UDP_GRO
+TEST_P(UdpListenerImplTest, UdpGroBasic) {
+  // We send 4 packets (3 of equal length and 1 as a trail), which are concatenated together by
+  // kernel supporting udp gro. Verify the concatenated packet is transformed back into individual
+  // packets
+  absl::FixedArray<std::string> client_data({"Equal!!!", "Length!!", "Messages", "trail"});
+
+  std::string stacked_message = absl::StrJoin(client_data, "");
+  client_.write(stacked_message, *send_to_addr_);
```

Comment:
Doing so, you are sending a single packet with concatenated payload. It doesn't matter because we are mocking recvmsg, but it is confusing to read the test. Either change it to send 4 packets, or comment why it doesn't matter.

Reply:
I have now changed it to sending 4 packets from the client. Also I added a comment describing the use of the stacked_message.